APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m

```diff
@@ -0,0 +1,293 @@
+@synthesize baseTrackingLocation = _baseTrackingLocation;
+@synthesize mostRecentUpdatedLocation = _mostRecentUpdatedLocation;
+- (NSArray*)locationDictionaryWithLocationManager:(CLLocationManager*)manager
+                       distanceFromReferencePoint:(CLLocationDistance)distanceFromReferencePoint
+                              andPreviousLocation:(CLLocation*)previousLocation
+{
+    NSString*   timestamp               = nil;
+    
+    if (manager.location.timestamp == nil)
+    {
+        timestamp = @"not available";
+    }
+    else
+    {
+        timestamp = manager.location.timestamp.toStringInISO8601Format;
+    }
+    
+    NSString*   distance                = [NSString stringWithFormat:@"%f", distanceFromReferencePoint];
+    NSString*   unit                    = @"meters"; //Hardcoded as Core Locations uses only meters
+    double      direction               = [previousLocation calculateDirectionFromLocation:manager.location];
+    NSString*   directionUnit           = @"radians"; //Not a user facing constant
+    
+    //  A negative value of manager.location.speed indicates an invalid speed.
+    NSString*   speed                   = nil;
+    
+    if (manager.location.speed >= 0)
+    {
+        double pace = manager.location.speed;
+        speed = [NSString stringWithFormat:@"%f", pace];
+    }
+    else
+    {
+        speed = NSLocalizedString(@"invalid speed", nil);
+    }
+
+    NSString*   speedUnit               = @"meters/second"; //Not a user facing constant
+    NSString*   floor                   = nil;
+    
+    //  A nil value of manager.location.floor indicates that this info is unavailable.
+    if (manager.location.floor == nil)
+    {
+        floor = NSLocalizedString(@"not available", nil);
+    }
+    else
+    {
+        NSInteger level = manager.location.floor.level;
+        floor = [NSString stringWithFormat:@"%ld", (long)level];
+    }
+    
+    //  Altitude of the location can be positive (above sea level) or negative (below sea level).
+    double      altitude                = manager.location.altitude;
+    NSString*   altitudeUnit            = @"meters";
+    NSString*   horizontalAccuracy      = [NSString stringWithFormat:@"%f", manager.location.horizontalAccuracy];
+    NSString*   horizontalAccuracyUnit  = @"meters";
+    NSString*   verticalAccuracy        = [NSString stringWithFormat:@"%f", manager.location.verticalAccuracy];
+    NSString*   verticalAccuracyUnit    = @"meters";
+    
+    return  @[timestamp, distance, unit, @(direction), directionUnit, speed, speedUnit, floor, @(altitude), altitudeUnit, horizontalAccuracy, horizontalAccuracyUnit, verticalAccuracy, verticalAccuracyUnit];
+}
+
+
+- (void)didRecieveUpdateWithLocationManager:(CLLocationManager*)manager withUpdateLocations:(NSArray*)locations
+{
+    __weak typeof(self) weakSelf = self;
+    
+    [self.healthKitCollectorQueue addOperationWithBlock:^
+    {
+        __typeof(self)  strongSelf  = weakSelf;
```

Comment:
You do this, and then access self directly. I think accessing self directly should be fine, so why are you doing this?